src/frontend/src/pages/ProjectPage/pages/ContractPage/pages/ManagePersonnelPage/hooks/useAddPersonnelForm.ts

```diff
@@ -0,0 +1,29 @@
+import { useCallback } from 'react';
+import Person from '../../../../../../../models/Person';
+import useForm from '../../../../../../../hooks/useForm';
+import { v1 as uuid } from 'uuid';
+
+const createDefaultState = (): Person[] => ([
+  {
+    personnelId:uuid(), 
+    name:"",
+    phoneNumber:"",
+    mail:"",
+    jobTitle:""
+  }]);
+
+const useAddPersonnelForm = (defaultState?: Person[] | null) => {
+    const validateForm = useCallback((formState: Person[]) => {
+        return formState
+        .find(p => 
```

Comment:
Isn't this the same as: `.find(p => p.name && p.phoneNumber && p.mail)`?  Would be easier to understand

Reply:
`p.name && p.phoneNumber && p.mail` would return something like `boolean | undefined` or `boolean | null` while wrapping the expression in `Boolean(...)` would ensure the boolean type and make it easier to use (especially with TypeScript)

Reply:
It would be easier to use, but we are not using it so it won't matter. The result from the `find()` would be the same either way. We could wrap the expression in a `Boolean(...)` or `!!(...)`, but that would only affect the readability.  But is the `Boolean(...) === false ? false : true` needed or could we just use `Boolean(...)?`

Reply:
The ternary is not necessary

Reply:
Find does not return a boolean. it returns the object. 
Find will here iterate until it finds a invalid line and then return it. 
Hence the ternery at the end. it equates the result from find to a boolean. 

If something is found, the form is invalid, else its valid.  
guess I could change it to;
`return  !formState.find(p => Boolean(p.name && p.phoneNumber && p.mail) === false) `

Reply:
If you want to return a boolean I would suggest using `some()` instead. `some() `returnes a boolean based on if a statement is true or false instead of a object like `find()` does. And if the intention is just want to check if a line is invalid, the ternary at the end is by my opinion not needed. Wouldn't something like this work? `formState.some(p => !(p.name && p.phoneNumber && p.mail))`

Reply:
Note: I now see that the ` ? false : true` ternary is **outside** the `find()`, my bad 😅. But my suggestion above still stands

Reply:
wasnt aware some() returned a bool on first hit. So that makes sense. 
The false check is still needed though in the end is still needed though. 
Cause I need to return false if something is found.